webrpc update from head

kipply opened the pull request.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/cohere-ai/webrpc?pullRequest=11) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/cohere-ai/webrpc?pullRequest=11) before we can accept your contribution.<br/>**0** out of **5** committers have signed the CLA.<br/><br/>:x: VojtechVitek<br/>:x: LukasJenicek<br/>:x: pkieltyka<br/>:x: klaidliadon<br/>:x: austinmilt<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/cohere-ai/webrpc?pullRequest=11) it.</sub>

VojtechVitek committed:
CI: Run Goreleaser on new git tag (#142)

Build webrpc-gen binary for multiple OS/archs.
Create a Github release with changelog & attach the binaries.

VojtechVitek committed:
Publish webrpc-gen as a Homebrew tap automatically (#144)

VojtechVitek committed:
Build and publish Docker image on new git tag (#145)

Build and publish webrpc-gen Docker image on new git tag

VojtechVitek committed:
README: Add Install steps, improve Getting started and Release notes (#146)

* Fix -version flag value built in CI

* README: Add Install steps & improve Getting started

* Goreleaser: Add Install steps to release notes

VojtechVitek committed:
webrpc-gen v0.8.0 - Add sort, trimPrefix and trimSuffix template functions (#147)

* Add trimPrefix and trimSuffix template functions

* Implement sort template function, add documentation

* Bump webrpc-gen minor version because of template API change

* Re-generate examples

VojtechVitek committed:
webrpc-gen v0.8.0: Implement template arrays (#149)

* Split Template API functions to logical groups; Update docs

* Implement array and append template functions

* Add tests for array functions

VojtechVitek committed:
Improve release notes (#150)

VojtechVitek committed:
Examples: Print available commands when invoking make (#152)

dependabot[bot] committed:
Bump minimatch from 3.0.4 to 3.1.2 in /_examples/node-ts/webapp (#151)

Bumps [minimatch](https://github.com/isaacs/minimatch) from 3.0.4 to 3.1.2.
- [Release notes](https://github.com/isaacs/minimatch/releases)
- [Commits](https://github.com/isaacs/minimatch/compare/v3.0.4...v3.1.2)

---
updated-dependencies:
- dependency-name: minimatch
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

pkieltyka committed:
print summary after code generation (#155)

* print summary after code generation

* typo

pkieltyka committed:
update print summary layout

dependabot[bot] committed:
Bump decode-uri-component from 0.2.0 to 0.2.2 in /_examples/node-ts/webapp (#156)

Bump decode-uri-component in /_examples/node-ts/webapp

Bumps [decode-uri-component](https://github.com/SamVerschueren/decode-uri-component) from 0.2.0 to 0.2.2.
- [Release notes](https://github.com/SamVerschueren/decode-uri-component/releases)
- [Commits](https://github.com/SamVerschueren/decode-uri-component/compare/v0.2.0...v0.2.2)

---
updated-dependencies:
- dependency-name: decode-uri-component
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump qs from 6.5.2 to 6.5.3 in /_examples/node-ts/webapp (#158)

Bumps [qs](https://github.com/ljharb/qs) from 6.5.2 to 6.5.3.
- [Release notes](https://github.com/ljharb/qs/releases)
- [Changelog](https://github.com/ljharb/qs/blob/main/CHANGELOG.md)
- [Commits](https://github.com/ljharb/qs/compare/v6.5.2...v6.5.3)

---
updated-dependencies:
- dependency-name: qs
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump express from 4.16.4 to 4.17.3 in /_examples/node-ts/server (#159)

Bumps [express](https://github.com/expressjs/express) from 4.16.4 to 4.17.3.
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/master/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.16.4...4.17.3)

---
updated-dependencies:
- dependency-name: express
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Commit:
Update testing suite (#157)

Test Interoperability

- Generate new go server listening on 9999 port satisfying api schema
defined in tests/schema/api.ridl

- Generate go client and server during build step since it's required by
webrpc-server binary

- Clean test cache before each run

- Test interoperability -> getting data from server, sending back and
testing if everything stay the same

- Implement shutdown http method on server to give client possibility of
closing down the server

VojtechVitek committed:
Update the new testing suite (#161)

* Make all tests use httptest.NewServer()

- Remove `make test' running test server binary in background
- The /shutdown endpoint was not 100% reliable based on my experience,
  the process still ran in the bacground and occupied port :9999 forever
- I propose to spin up the test server binary against a client binary
  in a separate test via os.Exec(), mimicking what generator repos will do

* webrpc-test -server -port=9988 -timeout=100s

* webrpc-test -client -url=http://localhost:9988

* Fix make build

VojtechVitek committed:
Go releaser: Build webrpc-test binary (#162)

* Go releaser: Build webrpc-test binary

* Try to fix goreleaser build

* Add brew tap for webrpc-test

* Improve Docker release docs

* Goreleaser brews must reference archive ids

* Goreleaser: Builds > Archives > Brews

VojtechVitek committed:
Github actions: Cache webrpc templates between CI jobs (#166)

VojtechVitek committed:
Relax template cache invalidation (#165)

- Fixes #164
- Refresh cache only if we can fetch all files from remote git
- Fall-back to old cache (over TTL) on git fetch error
- Fetch *.go.tmpl files only (was *.tmpl)
- Pre-fetch all template files when FS is initially loaded
- Print new webrpc-gen cache summary

$ webrpc-gen -schema=./tests/schema/api.ridl -target=golang -pkg=client -client -out=./tests/client/client.gen.go
=======================================
|      webrpc generated summary       |
=======================================
 webrpc-gen version : v0.8.x-dev
 target             : github.com/webrpc/gen-golang
 target cache       : /var/folders/23/vc4gzw8n52gc17yx4wfqfj3c0000gn/T/webrpc-cache/2568534563-gen-golang
 target cache age   : 1h51m3s (failed to refresh: get git repository: GET https://api.github.com/repos/webrpc/gen-golang: 403 API rate limit exceeded for 178.248.249.18. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.) [rate reset in 9m42s])
 schema file        : ./tests/schema/api.ridl
 output file        : ./tests/client/client.gen.go

VojtechVitek committed:
Github CI: Run client-server interoperability test (golang@latest <-> golang@latest) (#163)

* make test: Run client-server (both golang@latest) interoperability test

* Ensure tests & examples are ran against current ./bin/webrpc-gen

* Makefile supports exporting PATH globally

* Wait for test server to come up

* make test: Wait for test server to finish

* webrpc-test -server: Exit success on normal shutdown

* Run Go 1.19 before Go 1.16 tests

Before this change, the Go 1.19 test was canceled automatically
if the slower Go 1.16 test failed (which is more likely).

* make generate: Fix directory name in the output

VojtechVitek committed:
Internal renaming (#168)

Split off https://github.com/webrpc/webrpc/pull/134 by @pkieltyka.

VojtechVitek committed:
Move VERSION to top-level package (#169)

VojtechVitek committed:
v0.9.0: Fix non-deterministic schema hash (#172)

* Fix non-deterministic schema hash

Schema hash was generated from schema.JSON(nil) including Imports
pointing at absolute paths. The CI generated different hash compared
to my development setup when there were RIDL imports involved.

1. Can we simply disable the Imports field from Schema JSON?
2. Do we need to keep schema.Imports around at all?

* Remove schema.Imports

LukasJenicek committed:
Improve testing suite (#174)

* Fix typo

* Replace sleep with netcat to check if server is ready

VojtechVitek committed:
v0.9.0: Rename RIDL "message" to "struct", schema "messages" to "types" (#171)

* RIDL: Rename "message" to "struct"

Split off https://github.com/webrpc/webrpc/pull/134 by @pkieltyka

* Schema: Rename "messages" to "types"

Split off https://github.com/webrpc/webrpc/pull/134 by @pkieltyka

* Fix tests

* Update .json schemas & provide migrate.js script

* Fix generating templates

* Regenerate examples

* Re-add two template functions to docs

LukasJenicek committed:
Allow webrpc-test binary to print out ridl scheme (#176)

Allow test binary to print out ridl scheme

VojtechVitek committed:
Fix CI: Regenerate all _examples/ (#177)

* Fix CI: Regenerate all _examples/

Turns out "go generate ./.." didn't regenerate _examples/ directory because of the leading underscore character.

* make generate

VojtechVitek committed:
Invalidate Github Actions cache every day (#180)

Github cache eviction policy:
- GitHub will remove any cache entries that have not been accessed in over 7 days.

VojtechVitek committed:
v0.9.0 changelog: Document ENUM type JSON schema changes (#179)

VojtechVitek committed:
Accept io.FS in RIDL parser (#178)

* Remove unused schema from templateFuncMap()

* Accept io.FS in RIDL parser

Fixes #173

* Simplify reading JSON schema file

* Remove invalid wrapping of nil error

* Remove ridl.mockImport

* Fix building all tests

* Don't regenerate golden file on each test run

* Improve RIDL parser tests

* TestRIDLImports(): Use fstest.MapFS, test import ../parent.ridl

* TestRIDLImportsExampleDir(): Use os.DirFS() to test _example/ dir

* TestRIDLImports(): Test imported struct types

VojtechVitek committed:
Improve interoperability test suite: Add more RPC methods to TestApi (#181)

* Improve interoperability test server

* Rename tests/schema/api.ridl to test.ridl

* Rename ComplexApi to TestApi

* Implement new TestApi RPC methods

* Define fixtures, use cpm.Equal() and cpm.Diff() for comparison

* Reverse cmp.Diff() arguments to make diff logical

* Fix interoperability tests by using float64 in fixtures

* Fix webrpc-test build

VojtechVitek committed:
Bump to webrpc@v0.10.0 (#182)

* Bump to webrpc@v0.10.0

* make generate

* Document Interoperability tests in generators docs

* Document v0.10.0 breaking changes in CHANGELOG

* Add table of contents to CHANGELOG

dependabot[bot] committed:
Bump json5 from 1.0.1 to 1.0.2 in /_examples/node-ts/webapp (#184)

* Bump json5 from 1.0.1 to 1.0.2 in /_examples/node-ts/webapp

Bumps [json5](https://github.com/json5/json5) from 1.0.1 to 1.0.2.
- [Release notes](https://github.com/json5/json5/releases)
- [Changelog](https://github.com/json5/json5/blob/main/CHANGELOG.md)
- [Commits](https://github.com/json5/json5/compare/v1.0.1...v1.0.2)

---
updated-dependencies:
- dependency-name: json5
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

* Update to javascript@v0.10.0

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Vojtech Vitek <vojtech.vitek@golang.cz>

VojtechVitek committed:
gen/README.md: Fix join() template function documentation (#185)

VojtechVitek committed:
Remove unnecessary schema.VarName type (#186)

* Remove unused schema.VarName methods

* Remove unnecessary schema.VarName type

It was difficult to work with schema.VarName (custom string type) in Go templates.
Calling `{{ $type.Name }}` failed with the following error:
    ```
    command failed: template: struct.go.tmpl:9:17: executing "struct" at <.Name>: wrong type for value; expected string; got schema.VarName
    ```

This change will remove type-casting via `{{ printf "%v" $typeName }}` from templates and improve readability.

VojtechVitek committed:
Define custom errors in RIDL (#187)

* Rename schema.go to embed.go

* Custom errors: Define API errors in RIDL

Split off "v2" branch by @pkieltyka.

Originally introduced in
https://github.com/webrpc/webrpc/tree/schema-v2
https://github.com/webrpc/webrpc/pull/134
https://github.com/webrpc/webrpc/pull/123

* Regenerate examples

VojtechVitek committed:
Add interoperability RIDL<->JSON golden test file (#188)

* Add golden JSON schema file test to tests/

* Improve schema.ToJSON()

- Remove optional parameter from the schema pkg API
- Always pretty print JSON schema
- Golden files changes will be more readable
- Schema hash will be still predictable
- Improve performance via buf.String() on stack

* Regenerate examples/tests

* Remove unused schema.HasFieldType() method

VojtechVitek committed:
Fix clean GH Actions cache job (#189)

VojtechVitek committed:
Update the RIDL error syntax & parser (#191)

* Remove noisy log in TestSyntaxError()

* Update the RIDL error syntax & parser

error <code> <name> <message> [HTTP <status code>]

VojtechVitek committed:
Fix default error HTTP status code, add more RIDL tests (#192)

* Add RIDL tests for custom errors

* Default HTTP status code should be 400

VojtechVitek committed:
gen/README.md: Document schema errors, fix link (#194)

VojtechVitek committed:
Implement interoperability tests for schema errors (v0.11.0) (#193)

* Implement interoperability tests for schema errors

* Update test RIDL to be backward compatible with legacy errors

* Regenerate interoperability tests with github.com/webrpc/gen-golang@tags/errors

* Make httpStatus a required field

VojtechVitek committed:
Add Go 1.20 to CI matrix (#195)

* Add Go 1.20 to CI matrix

* Go 1.20 must be string, so yaml doesn't round it to 1.2

VojtechVitek committed:
Add built-in webrpc errors (#196)

* Add built-in webrpc errors

* Regenerate interoperability tests

VojtechVitek committed:
Schema errors: HTTP status code must be between 400-599 (#197)

If someone defined HTTP 200 for a schema error, the webrpc
clients wouldn't recognize it as an error. Instead, they'd
treat the response body as a regular response data object.

VojtechVitek committed:
Add interoperability test for error cause (#198)

VojtechVitek committed:
Bump to v0.11.0

VojtechVitek committed:
Bump to v0.11.0 - regenerate examples (#199)

VojtechVitek committed:
Update links to migration guide

VojtechVitek committed:
Update to latest Go and TS gens with schema support (#200)

Update to latest Go and TS gens with schema errors support

VojtechVitek committed:
Add release notes / changelog for v0.11.0 (#201)

* Add release notes / changelog for v0.11.0

* Regenerate using latest golang@v0.11.0

* Improve .md formatting

VojtechVitek committed:
Bump version to v0.12.x-dev

VojtechVitek committed:
Regenerate examples

pieter-lazzaro committed:
Fix windows path handling (#202)

* fs.FS always uses "/" as the separator

* Account for Windows volumes

---------

Co-authored-by: Pieter Lazzaro <pieter.lazzaro@pureharvest.com.au>

VojtechVitek committed:
Regenerate examples (latest TypeScript example) (#207)

Follow-up for #202

VojtechVitek committed:
Add `webrpc-test -printJSON' (#206)

VojtechVitek committed:
Implement `webrpc-test -waitForServer -timeout=10s' (#208)

* Implement `webrpc-test -waitForServer -timeout=10s'

Will remove the need for waiting for the test server readiness via custom scripts, ie.
https://github.com/webrpc/gen-typescript/blob/a80789817830a8e55480745111eb259da5eeedf6/tests/test.sh#L13-L14

* PR feedback from @holowcost, add delay & use select{}

* PR feedback: Remove unnecessary os.Exit(0)

LukasJenicek committed:
Fix examples after introducing customer webrpc errors (#209)

* fix undefined error by defining custom error when user is not found

* improve running node js client -> go server communication example

* generate server & client with latest webrpc-gen version

VojtechVitek committed:
Template generator: Implement generic array functions (#210)

* Template generator: Implement generic array functions

* Update join(), first(), last() to work with both []string and []any

dependabot[bot] committed:
Bump semver from 5.7.0 to 5.7.2 in /_examples/node-ts/webapp (#212)

Bumps [semver](https://github.com/npm/node-semver) from 5.7.0 to 5.7.2.
- [Release notes](https://github.com/npm/node-semver/releases)
- [Changelog](https://github.com/npm/node-semver/blob/v5.7.2/CHANGELOG.md)
- [Commits](https://github.com/npm/node-semver/compare/v5.7.0...v5.7.2)

---
updated-dependencies:
- dependency-name: semver
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

pkieltyka committed:
cosmetic, errors should be defined before services

pkieltyka committed:
add Unauthorized and PermissionDenied types in golang-basics example

pkieltyka committed:
example (golang-basics): demo how to use ErrorWithCause(..)

VojtechVitek committed:
Update to latest typescript@v0.11.2 generator (#217)

VojtechVitek committed:
Remove github.com/pkg/errors (#216)

VojtechVitek committed:
Fix duplicate or invalid json fields (#218)

* Fix duplicate or invalid json fields

Fixes #66

* Fix golang-nodejs example: Use go.tag.json instead of json

* Acknowledge the special `json = -` behavior

* Fix hello-webrpc and hello-webrpc-ts examples

* Fix tests with duplicate json fields

* Update to latest javascript@v0.11.0 generator

pkieltyka committed:
errors, add WebrpcInternalError and set HTTPStatus 400 default on ErrWebrpcRequestFailed

VojtechVitek committed:
make generate

VojtechVitek committed:
Bump to v0.13.0-dev (#220)

VojtechVitek committed:
webrpc-gen: Add -target=json CLI flag (#219)

Works well with the -out= flag, same as other targets.
Removes the need for the special -test flag.

VojtechVitek committed:
Add v0.12.0 changelog

VojtechVitek committed:
Fix panic

VojtechVitek committed:
Generate .json schema in tests folder

VojtechVitek committed:
Implement dump template function (#221)

1. Useful for debugging template vars as comments in the generated code:
/* {{- dump .TemplateVar -}} */

2. Useful for debugging template vars in webrpc-gen stderr output:
{{- stderrPrintf (dump .) -}}
{{- exit 1 -}}

VojtechVitek committed:
Implement hasField template function (#222)

VojtechVitek committed:
Generate code to the output file even on formatting error (#223)

This is very useful when developing/debugging templates that render invalid code.

VojtechVitek committed:
Implement stderrPrint template function, suppress <nil> output (#224)

Implement stderrPrint template function, suppress <nil> output from stderrPrintf

VojtechVitek committed:
Add -target=debug to print schema object incl. Go type information (#225)

Add -target=debug to print schema and template vars

VojtechVitek committed:
Make -target=debug output idempotent, track tests schema (#226)

VojtechVitek committed:
Indent -target=debug output with tabs for better readability

VojtechVitek committed:
webrpc-gen: Print custom targets properly

VojtechVitek committed:
Update to golang@v0.12.5: Simplify Enums to TextMarshalers (#228)

pkieltyka committed:
ridl rpc succinct form (#229)

* alias type and idl rpc succinct method format

* passing tests

* ridl succinct form

* revert back to string from VarName

* revert back to string from VarName

VojtechVitek committed:
Bump to v0.14.0-dev

VojtechVitek committed:
Regenerate examples

VojtechVitek committed:
Update to golang@v0.13.1 (#230)

Update to golang@v0.13.1

- https://github.com/webrpc/gen-golang/pull/31
- https://github.com/webrpc/gen-golang/pull/36
- https://github.com/webrpc/gen-golang/pull/35
- https://github.com/webrpc/gen-golang/pull/34
- https://github.com/webrpc/gen-golang/pull/33
- https://github.com/webrpc/gen-golang/pull/32
- https://github.com/webrpc/gen-golang/pull/37

VojtechVitek committed:
Update to golang@v0.13.3 (#234)

VojtechVitek committed:
Regenerate using latest golang@v0.13.7 template (#240)

VojtechVitek committed:
Remove obsolete example .gen.go file (#238)

VojtechVitek committed:
Remove the use of obsolete io/ioutil pkg in RIDL tests (#239)

VojtechVitek committed:
Add missing go.mod to golang-imports example (#241)

VojtechVitek committed:
RIDL parser: Error out if stream is defined per argument type (#242)

* RIDL parser: Error out if stream is defined per argument type

* Improve error message & wrap underlying error

VojtechVitek committed:
Define new webrpc errors (#244)

* Define new webrpc errors

* Regenerate examples

LukasJenicek committed:
Support enum strings (#247)

* allow string enum

* update go example with enum string support

* do not return error when type is not enum

* remove error out on structType func

LukasJenicek committed:
regenerate examples with latest gen-golang (#250)

LukasJenicek committed:
do not create cache directory if there is 0 templates (#249)

LukasJenicek committed:
get all required fields (#253)

LukasJenicek committed:
parse comments and store them to webrpc.schema (#252)

* parse comments and store them to webrpc.schema

* use single comments field instead of struct

* remove unused type

* create the comment line during tokenize process

LukasJenicek committed:
Add replaceAll function to template functions (#257)

* add strings replace all function

* bump version and regenerate examples

LukasJenicek committed:
parse multiline comments 2 lines and more (#258)

* fix referencing enums

* parse multiline comments 2 lines and more

* skip reading comments once we already hit one comment and there is empty line

LukasJenicek committed:
bump webrpc version (#259)

VojtechVitek committed:
Add Kotlin client generator (#260)

https://github.com/webrpc/gen-kotlin

VojtechVitek committed:
Regenerate examples with latest golang@v0.14.1 (#261)

VojtechVitek committed:
Move .goreleaser outside of Github workflows folder (#262)

Fixes false positive errors in Github Actions UI.

VojtechVitek committed:
Infer webrpc version from Go module or git history, if not set (#263)

* Infer webrpc version if imported from another Go module

Useful if you import webrpc into your Go project and invoke it via
go run github.com/webrpc/webrpc/cmd/webrpc-gen.

* Infer version from git as a fallback

* make build: Inject version into the Go binary

* Replace webrpc version in all generated files to avoid git conflicts

* Update to actions/checkout@v4; Use 20 commit depth for git tags

* Improve make generate diff

VojtechVitek committed:
Fix actions/checkout@v4 git clone issue on push tags

fatal: Cannot fetch both 8c823bfe0180c56c89e9a54148c196fb1df3ba5b and refs/tags/v0.15.3 to refs/tags/v0.15.3

https://github.com/actions/checkout/issues/1467

VojtechVitek committed:
Fix .goreleaser path

VojtechVitek committed:
Fix version: git describe --tags (#265)

* Fix version: git describe --tags

Without --tags option, git describe returns only annotated tags..

* Regenerate with latest golang template

VojtechVitek committed:
Embed default templates into webrpc-gen via Go modules (#266)

* Embed golang template via Go module

* Embed typescript, javascript, openapi and kotlin templates

* Regenerate examples

dependabot[bot] committed:
Bump express from 4.17.3 to 4.19.2 in /_examples/node-ts/server (#264)

Bumps [express](https://github.com/expressjs/express) from 4.17.3 to 4.19.2.
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/master/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.17.3...4.19.2)

---
updated-dependencies:
- dependency-name: express
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

VojtechVitek committed:
Use embedded templates for cached git URLs, improve webrpc-gen help (#269)

* Use embedded templates for cached git URLs, improve webrpc-gen help

* Support older versions of Go

* Bump CI to Go 1.22

* Print full template git repo tag

pkieltyka committed:
internal comments (#268)

* internal comments

* ignore comments starting with bang

---------

Co-authored-by: lukasjenicek <lukas.jenicek5@gmail.com>

pkieltyka committed:
README, mention vscode plugin for RIDL syntax highlighting

LukasJenicek committed:
fix parsing comments with http scheme and fqdn (#270)

LukasJenicek committed:
use different url in tests (#271)

LukasJenicek committed:
keep empty lines while parsing comments (#272)

VojtechVitek committed:
Webrpc SSE streaming examples (#237)

* Improve error message & wrap underlying error

* Remove obsolete example .gen.go file

* Remove the use of obsolete io/ioutil pkg in RIDL tests

* RIDL parser: Error out if stream is defined per argument type

* WIP: Add Golang SSE streaming example

* A working protype of Go<->Go SSE streaming in NDJSON

* Improve/polish Go generated code

* Fix make generate, add helper targets

* wip: sse ts client

* add onclose

* add comments

* add option types

* add reusable sseResponse function

* fix type for headers

* Dont convert error to string

* Add generic to WebRpcSSEOptions.onMessage

* Add SubscribeMessagesReturn type

* Ignore empty lines

* trigger onError if sseResponse fails

* send abort signal on clicking disconnect button

* rename interface WebRpcSSEOptions to WebRpcStreamOptions

* onClose handled in sseResponse()

* return WebrpcRequestFailed for aborting

* dont throw error from sseResponse, remove the catch statement

* Implement Chat server example with msg broadcasting

* rename interfaces for convention

* tmp: abuse buildResponse to get error

* add vitest

* add missing return

* basic unit test setup

* Send ping() right away to unblock clients

* Send newline after webrpcError to comply with NDJSON format

* npm run test

* Improve callback debugging messages

* Re-introduce backend timeout

* Support custom webrpc errors

* Simulate network error / server shutting down

Run server with:
while true ; do go run ./; done

* add timeout to stream resonse

* fixe timeout value

* Add CORS, expect username on both endpoints

* Update TS contract

* Send "SYSTEM" messages when someone joins/leaves

* new design

* Remove proxy

* Fix typo in generated code

* Fix timestamp

* fix server port

* fix scroll overflow

* Use WebrpcError

* improve log style

* add custom message to abort error

* Update RIDL

* Fix typo

* Move the generated method up, so it's easier to templatize

* PR feedback

* Remove doJSONRequest() in favor of doHTTPRequest()

* Move streamWriter/streamReader definition to improve template readability

* Close the response body of client request

* Close request's response body properly & remove Read() for loop

* Handle context timeouts in Go client after each ping message

* Improve Go tests

* Define new webrpc errors

* Fix and improve generated code

* Fix RIDL definition

* generated with latest typescript templates

* backward compatible

* add logAppend helper

* change name to toggle connect

* limit height of log

* add dynamic text to button

* update connection status

* prettier

* add reconect

* add exponential backoff

* add less expensive way of checking for timout

* allways clear interval

* fix missing lastReadTime

* improve onError

* fix error handeling

* error if abort signal is attemted to reconnect

* remove error overload

* regenerated contract

* Dont swollow/recast onMessage errors

* Regenerate examples

---------

Co-authored-by: Alexander Kolberg <alexander.kolberg@alida.com>
Co-authored-by: Jakub Alkhaled <jackkaled@gmail.com>
Co-authored-by: Umitcan Bal <umitcanbal@gmail.com>
Co-authored-by: Lukas Jenicek <lukas.jenicek5@gmail.com>
Co-authored-by: Alexander Kolberg <alexander.kolberg@gmail.com>

dependabot[bot] committed:
Bump vite from 5.0.2 to 5.0.13 in /_examples/golang-sse/webapp (#273)

Bumps [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) from 5.0.2 to 5.0.13.
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/v5.0.13/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v5.0.13/packages/vite)

---
updated-dependencies:
- dependency-name: vite
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

VojtechVitek committed:
Update go.mod deps, bump to Go 1.19 (#274)

* Update go.mod deps, explicitly keep golang.org/x/sys at Go 1.16

* Let's drop support for Go 1.16, since golang.org/x/crypto fails to build too

Error: ../../../go/pkg/mod/golang.org/x/crypto@v0.22.0/ssh/cipher.go:19:2: //go:build comment without // +build comment
Error: ../../../go/pkg/mod/golang.org/x/crypto@v0.22.0/ssh/kex.go:19:2: //go:build comment without // +build comment

* Bump to Go 1.18 as minimal supported version

* Update examples to github.com/go-chi/chi@v5

* go mod tidy; update _example deps

* Use git describe --tags

* Bump to Go 1.19 to support golang.org/x/tools 2022+

VojtechVitek committed:
Remove reserved keyword in Dart (#276)

* change reserved keyword enum to status in test schema

* Fix tests

---------

Co-authored-by: Austin Milt <austin.w.milt@gmail.com>

austinmilt committed:
additional generator functions (#277)

* add gen lastIndex func

* add gen math funcs

* add gen indent and strRepeat funcs

* spelling

* use sprig v3 gen funcmap

austinmilt committed:
declare the timestamp standard (#279)

LukasJenicek committed:
skip comment when its 2 lines above (#280)

VojtechVitek committed:
Golang webrpc streaming client fix (#282)

* Fix `make curl' in golang-sse example

* Add handy go.mod replace directives

* Improve streaming tests

* Fix 10k test

* Update the Go template

* Update to golang@v0.14.4

* Update to golang@v0.14.5

VojtechVitek committed:
Update to golang@v0.14.7 and all other dependencies (Go 1.19 compatible) (#285)

* Update to golang@v0.14.6

* Update dependencies (compatible with Go 1.19)

go get -u go@1.19 ./...

* Regenerate examples

* make dep-upgrade-all: Require Go 1.19 for now

* Update to golang@v0.14.7

VojtechVitek committed:
Update to golang@v0.14.8 (#287)

* Update to golang@v0.14.8

* Regenerate tests/examples

VojtechVitek committed:
Embbed dart client into webrpc-gen (#288)

https://github.com/webrpc/gen-dart/releases/tag/v0.1.1

Kudos to @austinmilt for implementing https://github.com/webrpc/gen-dart/pull/3.

VojtechVitek committed:
Goreleaser: Try to fix changelog with github-native impl (#289)

https://goreleaser.com/customization/changelog/

klaidliadon committed:
Update gitfs dependencies using fork (#290)

update gitfs dependencies using fork

VojtechVitek committed:
Update to github.com/posener/gitfs@v1.2.2 (#291)

klaidliadon committed:
Enum Type and `listElemType` fix (#294)

* don't convert type to string

* enum type

* Support enum in map

* update tests

* move enum to another struct, add example

klaidliadon committed:
Add tests for new enum methods (#296)

add tests for new enum methods

dependabot[bot] committed:
Bump ws from 5.2.2 to 5.2.4 in /_examples/node-ts/webapp (#295)

Bumps [ws](https://github.com/websockets/ws) from 5.2.2 to 5.2.4.
- [Release notes](https://github.com/websockets/ws/releases)
- [Commits](https://github.com/websockets/ws/compare/5.2.2...5.2.4)

---
updated-dependencies:
- dependency-name: ws
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

VojtechVitek committed:
Fix GoReleaser (#297)

https://goreleaser.com/deprecations/#-rm-dist

```
--rm-dist has been deprecated in favor of --clean.
```

This will fix the GoReleaser pipeline in master.

VojtechVitek committed:
Update to goreleaser/goreleaser-action@v6 with v2 yaml (#298)

VojtechVitek committed:
Update to gen-javascript@v0.13.0, fix hello-webrpc example (#299)

* Update to gen-javascript@v0.13.0

go get github.com/webrpc/gen-javascript@v0.13.0

* Fix hello-webrpc example

LukasJenicek committed:
Service method annotations (#308)

* parse annotations for service methods

* add annotations to schema

* on deprecate hook when some of the endpoints been deprecated

* regenerate examples

* use request in deprecate hook

* finish annotation parsing

* report deprecated endpoint onRequest

* handle same annotation types separated by comma

* return error OnRequest

* renerate with latest gen-golang tmpl

* sort imports

* regenerate examples

* remove unused annotations from test case

* regenerate examples with latest gen-golang

* use latest gen-golang

* regenerate using latest gen-golang

* add test for deprecated user endpoint

* add onError hook to tests

* enable parsing annotation with multiple values

* regenerate using latest gen-golang

LukasJenicek committed:
update gen-golang to v0.15.0 (#312)

VojtechVitek committed:
Improve Makefile, auto-generate 'make all' (#313)

VojtechVitek committed:
update to gen-typescript@v0.14.1 (#315)

VojtechVitek committed:
Update deps

VojtechVitek committed:
Revert "Update deps"

This reverts commit 6e8be5b2055cae53c4a19a4210075665c5e8a648.

LukasJenicek committed:
Match services and ignore methods  with specific annotations (#314)

* implement service filtering and apply in golang-basics example

* filter out methods with specific annotation

* transform services to map internally

* match annotation key and value or just annotation type

* remove redundant whitespace trimming

LukasJenicek committed:
annotation parser improvements (#316)

improve annotation parser

LukasJenicek committed:
Add template version and template target to template vars (#310)

* add template version and template target to template vars

* regenerate tests

* add code gen header to schema

* rename fields on schema tmpl

LukasJenicek committed:
update openapi to latest version (#317)

LukasJenicek committed:
match specific annotations (#318)

LukasJenicek committed:
mention ridlfmt and add it to example (#321)

LukasJenicek committed:
render schema hash if schema version in ridl is empty (#322)

* render schema hash if schema version in ridl is empty

* version format - go dependency like

LukasJenicek committed:
Update versions of gen-golang and gen-typescript (#323)

* update gen-golang to v0.16.0

* update gen-typescript to v0.15.0

* regenerate all examples using latest tmpl versions

VojtechVitek committed:
Remove binary from source code (#326)

LukasJenicek committed:
update gen-typescript to v0.15.1 (#327)

VojtechVitek committed:
Update examples - add Webrpc header to CORS (#324)

* Update examples - add Webrpc header to CORS

* Expose Webrpc header to CORS clients

david-littlefarmer committed:
Update gen-golang to v0.16.1 (#328)

* gen-golang v0.16.0 -> v0.16.1

* examples

* tests

LukasJenicek committed:
bump gen-golang and gen-typescript to latest version (#329)

* bump gen-typescript to latest

* bump gen-golang to latest

VojtechVitek committed:
RIDL: Fix importing errors (#331)

* Add RIDL example: Import errors.ridl with error definitions

* Import RIDL file with errors in tests

* Fix importing RIDL errors

* Imports in the example2 test RIDL with golden file

* Fix stream RPC example (ridlfmt needs fix)

* Update golden file

VojtechVitek committed:
Update to golang@v0.17.0 and typescript@v0.16.1 (#333)

* Update to golang@v0.17.0 and typescript@v0.16.1

* Regenerate examples/tests

VojtechVitek committed:
Update to gen-typescript@v0.16.2 (#336)

* Update to gen-typescript@v0.16.2

https://github.com/webrpc/gen-typescript/releases/tag/v0.16.2

* make generate

VojtechVitek committed:
Upgrade to Go 1.23; support Go 1.22+; upgrade deps (#337)

* Upgrade deps

* Upgrade to Go 1.23; support Go 1.20+

* Require Go 1.22+, upgrade deps

* CI: Debug git describe --tags

* Is there a bug in actions/checkout@v4?

https://github.com/actions/checkout/issues/1781

* Fix go.sum

* Improve inferring webrpc-gen version from a Go module

VojtechVitek committed:
Update README: Support Go 1.22+

VojtechVitek committed:
Use strings.Cut() and .CutPrefix(), as we bumped Go version (#338)

VojtechVitek committed:
Update gen-typescript & other deps (#339)

* Update deps

* Update to gen-typescript@v0.16.3

* Regenerate examples

VojtechVitek committed:
go.mod: Remove replace directive (#340)

VojtechVitek committed:
Update to gen-golang@v0.18.0: Support alternative JSON encodings (#342)

VojtechVitek committed:
Update to gen-golang@v0.18.1: Opinionated Sonic config (#343)

dependabot[bot] committed:
Bump vite from 5.0.13 to 5.1.8 in /_examples/golang-sse/webapp (#305)

Bumps [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) from 5.0.13 to 5.1.8.
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/v5.1.8/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v5.1.8/packages/vite)

---
updated-dependencies:
- dependency-name: vite
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump serve-static from 1.13.2 to 1.16.0 in /_examples/node-ts/webapp (#304)

Bumps [serve-static](https://github.com/expressjs/serve-static) from 1.13.2 to 1.16.0.
- [Release notes](https://github.com/expressjs/serve-static/releases)
- [Changelog](https://github.com/expressjs/serve-static/blob/master/HISTORY.md)
- [Commits](https://github.com/expressjs/serve-static/compare/v1.13.2...1.16.0)

---
updated-dependencies:
- dependency-name: serve-static
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump webpack from 4.29.3 to 5.94.0 in /_examples/hello-webrpc-ts/webapp (#302)

Bumps [webpack](https://github.com/webpack/webpack) from 4.29.3 to 5.94.0.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v4.29.3...v5.94.0)

---
updated-dependencies:
- dependency-name: webpack
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump express from 4.19.2 to 4.20.0 in /_examples/node-ts/server (#306)

Bumps [express](https://github.com/expressjs/express) from 4.19.2 to 4.20.0.
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/master/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.19.2...4.20.0)

---
updated-dependencies:
- dependency-name: express
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump cross-spawn from 6.0.5 to 6.0.6 in /_examples/node-ts/webapp (#330)

Bumps [cross-spawn](https://github.com/moxystudio/node-cross-spawn) from 6.0.5 to 6.0.6.
- [Changelog](https://github.com/moxystudio/node-cross-spawn/blob/v6.0.6/CHANGELOG.md)
- [Commits](https://github.com/moxystudio/node-cross-spawn/compare/v6.0.5...v6.0.6)

---
updated-dependencies:
- dependency-name: cross-spawn
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump vitest from 1.0.2 to 1.6.1 in /_examples/golang-sse/webapp (#344)

Bumps [vitest](https://github.com/vitest-dev/vitest/tree/HEAD/packages/vitest) from 1.0.2 to 1.6.1.
- [Release notes](https://github.com/vitest-dev/vitest/releases)
- [Commits](https://github.com/vitest-dev/vitest/commits/v1.6.1/packages/vitest)

---
updated-dependencies:
- dependency-name: vitest
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump elliptic from 6.4.1 to 6.6.1 in /_examples/node-ts/webapp (#345)

Bumps [elliptic](https://github.com/indutny/elliptic) from 6.4.1 to 6.6.1.
- [Commits](https://github.com/indutny/elliptic/compare/v6.4.1...v6.6.1)

---
updated-dependencies:
- dependency-name: elliptic
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @babel/helpers from 7.4.3 to 7.26.10 in /_examples/node-ts/webapp (#348)

Bumps [@babel/helpers](https://github.com/babel/babel/tree/HEAD/packages/babel-helpers) from 7.4.3 to 7.26.10.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.26.10/packages/babel-helpers)

---
updated-dependencies:
- dependency-name: "@babel/helpers"
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

VojtechVitek committed:
Github action: make generate (#350)

* Github action: make generate

* actions/checkout@v4: Fetch git tags

VojtechVitek committed:
Update to gen-golang@v0.18.2 (#349)

* Update to gen-golang@v0.18.2
   * https://github.com/webrpc/gen-golang/releases/tag/v0.18.2
   * Opinionated config for -json=sonic by @VojtechVitek in https://github.com/webrpc/gen-golang/pull/84
   * Opinionated config for -json=jsoniter by @VojtechVitek in https://github.com/webrpc/gen-golang/pull/85
   * Improve import compatibility for sonic and jsoniter by @VojtechVitek in https://github.com/webrpc/gen-golang/pull/86

* [AUTOMATED] make generate

VojtechVitek committed:
Support JSDoc comments and @deprecated annotations (#352)

* Update to gen-golang@v0.18.3, gen-typescript@v0.17.0

* Add comments and @deprecated annotation to example RIDL files

* make generate

VojtechVitek committed:
Update to gen-openapi@v0.16.1 (#355)

Fixes bug with quotes that were not escaped, improves summary/description.

VojtechVitek committed:
Update to gen-openapi v0.16.2 (#356)

Groups service endpoints via OpenAPI tags

VojtechVitek committed:
Update to gen-golang@v0.18.4 to improves Sonic JSON encoding performance (#357)

* Update to gen-golang@v0.18.4 to improves Sonic JSON encoding performance

https://github.com/webrpc/gen-golang/releases/tag/v0.18.4

* make generate

david-littlefarmer committed:
gen-golang@v0.19.0 (#360)

* Update to gen-golang@v0.19.0

* make generate

klaidliadon committed:
Use a Graph to detect circular dependencies (#362)

VojtechVitek committed:
Update to openapi@v0.16.3 (#363)

klaidliadon committed:
Allow to import the same file across multiple files (#365)

* Make types available if imported by different files

* don't export absolute path in debug gen

* improve error

* use import error

* refactor importParser to simplify node and edge addition

* cache common imports

* fix test

* renmove unnecessary error

* add filename to parsed schema and update RIDL file handling

* update filename only if not empty

* refactor: simplify Parse method by removing unused index parameter

* Update schema/type.go

Co-authored-by: Vojtech Vitek (golang.cz) <vojtech.vitek@gmail.com>

* remove filename from schema

* update test debug

* remove oneliner

* refactor: improve variable handling in ParseSchemaFile function

* refactor: update NewParser to accept root path and adjust filename handling

* remove file and line form type, since we don't need them

---------

Co-authored-by: Vojtech Vitek (golang.cz) <vojtech.vitek@gmail.com>

VojtechVitek committed:
Update webrpc streaming example (chatbot) (#370)

* Rename _examples/golang-sse/ to _examples/webrpc-streaming

* Update webrpc streaming example

* Chat example: Implement reconnection logic and stream closed

* [AUTOMATED] make generate

* Handle chat server graceful shutdown (end of stream)

go get github.com/golang-cz/ringbuf@v0.0.5

VojtechVitek committed:
Rename ClientDisconnected error to ClientAborted (#371)

* Rename ClientDisconnected error to ClientAborted

* PR feedback

Thanks @AlexanderKolberg for the suggestion.

VojtechVitek committed:
Update to gen-golang@v0.20.0 and gen-typescript@v0.19.0 (#372)

* Update to gen-golang@v0.20.0 and gen-typescript@v0.18.0

go get github.com/webrpc/gen-golang@v0.20.0
go get github.com/webrpc/gen-typescript@v0.18.0

* Update to gen-typescript@v0.19.0

VojtechVitek committed:
Remove a file that was committed by mistake (#375)

VojtechVitek committed:
webrpc-streaming example: Update NPM deps (#376)

VojtechVitek committed:
Fix golang-nodejs example, update deps (#377)

* Fix golang-nodejs example, update deps

* [AUTOMATED] make generate

VojtechVitek committed:
Fix golang-nodejs example, update deps (#379)

* Install webify as a Go tool

* Update README, fix URL

* Update Go packages

* Fix status code expectation

VojtechVitek committed:
hello-webrpc-ts example: Replace obsolete webpack-dev-server with Vite (#380)

* Remove obsolete webpack-dev-server

* Replace webpack-dev-server with Vite, remove obsolete imports & upgrade deps

* Make the example server build

* [AUTOMATED] make generate

VojtechVitek committed:
Update to Go 1.25, require Go >= 1.23 (#381)

dependabot[bot] committed:
Bump vite from 4.5.14 to 5.4.20 in /_examples/hello-webrpc-ts/webapp (#382)

Bumps [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) from 4.5.14 to 5.4.20.
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/v5.4.20/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v5.4.20/packages/vite)

---
updated-dependencies:
- dependency-name: vite
  dependency-version: 5.4.20
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

VojtechVitek committed:
Fix chunk-encoding issue in webrpc streams (#387)

* Fix chunk-encoding issue in webrpc streams

Update to gen-golang@v0.21.0. Kudos to @kalalannn for the report, investigation & help with Go templates.

https://github.com/webrpc/gen-golang/releases/tag/v0.21.0

* [AUTOMATED] make generate

pieter-lazzaro committed:
Fix windows path handling (#384)

Co-authored-by: Pieter Lazzaro <pieter.lazzaro@pureharvest.com.au>

pkieltyka committed:
schema and ridl update for succinct form (#388)

pkieltyka committed:
update examples and go.mod to latest gen-golang

pkieltyka committed:
Revert "update examples and go.mod to latest gen-golang"

This reverts commit 3f7509ccf8305ab372882f916c8f112c5ec867ef.

pkieltyka committed:
re-gen schemas and examples based on latest version of gen-golang (#389)

corbanbrook committed:
method "errors" keyword (#383)

pkieltyka committed:
upgrade gen-typescript and re-gen all examples (#392)

pkieltyka committed:
update gen-golang, and add new funcmap firstWordToLower for templates (#393)

pkieltyka committed:
bigint support (#396)

pkieltyka committed:
re-gen examples

pkieltyka committed:
use latest tagged gen-golang and gen-typescript deps

pkieltyka committed:
upgrade gen-golang dep (#397)

* upgrade gen-golang dep

* [AUTOMATED] make generate

---------

Co-authored-by: pkieltyka <18831+pkieltyka@users.noreply.github.com>

AlexanderKolberg committed:
Add comment about min supported ES version (#398)

pkieltyka committed:
upgrade gen-typescript dep, and nil check fix with succinct form

pkieltyka committed:
update examples gen outputs

VojtechVitek committed:
Update to gen-typescript@v0.22.3 and gen-openapi@v0.16.4 (#400)

* Update to gen-typescript@v0.22.3 and gen-openapi@v0.16.4

* [AUTOMATED] make generate

VojtechVitek committed:
Update deps (#401)

VojtechVitek committed:
Update to gen-typescript@v0.22.4 to hotfix Webrpc header key (#402)

* Update to gen-typescript@v0.22.4 to hotfix Webrpc header key

* [AUTOMATED] make generate

VojtechVitek committed:
Update to gen-typescript@v0.22.5 (#403)

* Update to gen-typescript@v0.22.5

* [AUTOMATED] make generate

VojtechVitek committed:
Add new -schemaVersion= option to override schema version (#404)

Support $WEBRPC_SCHEMA_VERSION env var too, it's very useful when using  webrpc-gen from within //go:generate directives.

VojtechVitek committed:
Remove backward-compatibility with old webrpc-gen v0.6.0 templates (#405)

VojtechVitek committed:
Remove compatibility with deprecated v0.8.0 templates (#406)

dependabot[bot] committed:
Bump hono from 4.10.1 to 4.10.3 in /_examples/node-ts/server-hono (#399)

Bumps [hono](https://github.com/honojs/hono) from 4.10.1 to 4.10.3.
- [Release notes](https://github.com/honojs/hono/releases)
- [Commits](https://github.com/honojs/hono/compare/v4.10.1...v4.10.3)

---
updated-dependencies:
- dependency-name: hono
  dependency-version: 4.10.3
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

VojtechVitek committed:
Update to gen-golang@v0.23.2 (#407)

* Update to gen-golang@v0.23.2

* Update to gen-golang@v0.23.2

* make generate

pkieltyka committed:
upgrade gen-typescript

pkieltyka committed:
add -methodTreeShake flag (#411)

* add -methodTreeShake flag

* update

VojtechVitek committed:
Update to gen-golang@v0.24.0 and gen-openapi@v0.16.5 (#418)

* Update to gen-golang@v0.24.0 and gen-openapi@v0.16.5

* make generate

* Simplify make generate command

* Update README

* Simplify example makefiles; use local webrpc-gen bin

dependabot[bot] committed:
Bump github.com/go-chi/chi/v5 from 5.0.12 to 5.2.2 in /_examples/node-ts/server-go (#419)

Bump github.com/go-chi/chi/v5 in /_examples/node-ts/server-go

Bumps [github.com/go-chi/chi/v5](https://github.com/go-chi/chi) from 5.0.12 to 5.2.2.
- [Release notes](https://github.com/go-chi/chi/releases)
- [Changelog](https://github.com/go-chi/chi/blob/master/CHANGELOG.md)
- [Commits](https://github.com/go-chi/chi/compare/v5.0.12...v5.2.2)

---
updated-dependencies:
- dependency-name: github.com/go-chi/chi/v5
  dependency-version: 5.2.2
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump hono from 4.10.3 to 4.11.7 in /_examples/node-ts/server-hono (#417)

Bumps [hono](https://github.com/honojs/hono) from 4.10.3 to 4.11.7.
- [Release notes](https://github.com/honojs/hono/releases)
- [Commits](https://github.com/honojs/hono/compare/v4.10.3...v4.11.7)

---
updated-dependencies:
- dependency-name: hono
  dependency-version: 4.11.7
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

VojtechVitek committed:
Update to gen-openapi@v0.17.0 (#421)

VojtechVitek committed:
Update to gen-openapi v0.17.1 (#422)

dependabot[bot] committed:
Bump github.com/go-chi/chi/v5 from 5.1.0 to 5.2.2 in /_examples/webrpc-streaming (#420)

Bump github.com/go-chi/chi/v5 in /_examples/webrpc-streaming

Bumps [github.com/go-chi/chi/v5](https://github.com/go-chi/chi) from 5.1.0 to 5.2.2.
- [Release notes](https://github.com/go-chi/chi/releases)
- [Changelog](https://github.com/go-chi/chi/blob/master/CHANGELOG.md)
- [Commits](https://github.com/go-chi/chi/compare/v5.1.0...v5.2.2)

---
updated-dependencies:
- dependency-name: github.com/go-chi/chi/v5
  dependency-version: 5.2.2
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

VojtechVitek committed:
Quick guide for LLMs (#423)

VojtechVitek committed:
Move service definition to top of RIDL examples (#425)

* Move service definition to top of RIDL examples

* Fix golden file

* [AUTOMATED] make generate

VojtechVitek committed:
Add new "basepath" keyword to define base path for the API routes (#426)

* Add new "basepath" keyword to define base path for the API routes

This will allow users to
- override the default "/rpc" prefix
- use a custom prefix such as "/v1" to enable schema versioning

If empty, the value defaults to "/rpc" for backward compatibility.
However, "basepath" will be required in the future versions of webrpc.

* Update golden files

VojtechVitek committed:
Support basepath keyword in all generators (#427)

* Update golang, typescript, javascript, openapi, dart, kotlin

* Add basepath to README and examples

* make generate

* Update go.mod

VojtechVitek committed:
Enable replacement of existing artifacts in release

dependabot[bot] committed:
Bump @hono/node-server from 1.19.5 to 1.19.10 in /_examples/node-ts/server-hono (#432)

Bump @hono/node-server in /_examples/node-ts/server-hono

Bumps [@hono/node-server](https://github.com/honojs/node-server) from 1.19.5 to 1.19.10.
- [Release notes](https://github.com/honojs/node-server/releases)
- [Commits](https://github.com/honojs/node-server/compare/v1.19.5...v1.19.10)

---
updated-dependencies:
- dependency-name: "@hono/node-server"
  dependency-version: 1.19.10
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump hono from 4.11.7 to 4.12.4 in /_examples/node-ts/server-hono (#431)

Bumps [hono](https://github.com/honojs/hono) from 4.11.7 to 4.12.4.
- [Release notes](https://github.com/honojs/hono/releases)
- [Commits](https://github.com/honojs/hono/compare/v4.11.7...v4.12.4)

---
updated-dependencies:
- dependency-name: hono
  dependency-version: 4.12.4
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/cloudflare/circl from 1.6.1 to 1.6.3 (#430)

Bumps [github.com/cloudflare/circl](https://github.com/cloudflare/circl) from 1.6.1 to 1.6.3.
- [Release notes](https://github.com/cloudflare/circl/releases)
- [Commits](https://github.com/cloudflare/circl/compare/v1.6.1...v1.6.3)

---
updated-dependencies:
- dependency-name: github.com/cloudflare/circl
  dependency-version: 1.6.3
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

klaidliadon committed:
Bump github.com/webrpc/gen-golang from v0.25.0 to v0.26.0 (#433)

* Bump github.com/webrpc/gen-golang from v0.25.0 to v0.26.0

* [AUTOMATED] make generate

* Bump github.com/webrpc/gen-golang from v0.26.0 to v0.26.1

* [AUTOMATED] make generate

dependabot[bot] committed:
Bump hono from 4.12.4 to 4.12.7 in /_examples/node-ts/server-hono (#434)

Bumps [hono](https://github.com/honojs/hono) from 4.12.4 to 4.12.7.
- [Release notes](https://github.com/honojs/hono/releases)
- [Commits](https://github.com/honojs/hono/compare/v4.12.4...v4.12.7)

---
updated-dependencies:
- dependency-name: hono
  dependency-version: 4.12.7
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

pkieltyka committed:
improve enum string generation (#435)

* support dots in enum strings

* update

* bump gen-golang and gen-typescript

pkieltyka committed:
methodTreeShake flag to exclude types with no method reference

VojtechVitek committed:
Rename to SKILL.md

klaidliadon committed:
Fix succinct flag when only one side uses inline struct (#437)

* Enforce succinct form requirement for both input and output in method definitions

* regenerate examples and tests

* remove sed backup files

* reject mismatched succinct: require both inputs and outputs

klaidliadon committed:
add workflow to auto-bump generator dependencies (#438)

* add workflow to auto-bump generator dependencies

* [AUTOMATED] make generate

klaidliadon committed:
ci: improve bump-generator workflow

Two triggers:
- repository_dispatch: generator repos notify on new tag, bumps the
  specific generator immediately
- workflow_dispatch: manual fallback that polls all generators for
  updates by comparing go.mod against latest tags

Uses 0xsequence/actions/git-commit to commit changes and open a PR
on the auto/bump-generators branch. Multiple generator bumps
accumulate in the same PR.

Requires WEBRPC_DISPATCH_TOKEN org-level secret.

klaidliadon committed:
feat: add type alias support to RIDL parser and schema (#439)

* Add type alias support to RIDL parser and schema

Introduces a new `type Name: BaseType` syntax for defining type aliases
in RIDL schemas. Aliases support metadata annotations (`+ key = value`)
and can be used as struct fields, enum values, and method argument types.

* regenerate test debug golden file

The new Alias field on VarType changes the debug dump output.

klaidliadon committed:
fix(ci): use GITHUB_TOKEN for bump-generator workflow

klaidliadon committed:
fix(ci): improve bump-generator commit message format

github-actions[bot] committed:
bump generators (#440)

* bump gen-golang to v0.27.1

* [automated]: update gen-golang to v0.28.0
[automated]: update gen-typescript to v0.26.0
[automated]: update gen-javascript to v0.14.0
[automated]: update gen-kotlin to v0.2.0
[automated]: update gen-openapi to v0.18.0

---------

Co-authored-by: VojtechVitek <139342+VojtechVitek@users.noreply.github.com>
Co-authored-by: klaidliadon <5322228+klaidliadon@users.noreply.github.com>

github-actions[bot] committed:
[automated] Generators update (#451)

[automated]: update gen-kotlin to v0.3.0

Co-authored-by: VojtechVitek <139342+VojtechVitek@users.noreply.github.com>

tolgahan-arikan committed:
docs: update Kotlin generator description (#452)

* docs: update Kotlin generator description

* docs: trim Kotlin generator description

* docs: trim Kotlin README table spacing

klaidliadon committed:
fix(ridl): deduplicate errors on diamond import (#456)

github-actions[bot] committed:
[automated] Generators update (#454)

Co-authored-by: VojtechVitek <139342+VojtechVitek@users.noreply.github.com>
